lib/ansible/modules/cloud/amazon/code_pipeline.py

```diff
@@ -0,0 +1,293 @@
+      type: list
+    version:
+      description: THe version number of the pipeline. This number is auto incremented when pipeline params are changed.
+      returned: always
+      type: int
+'''
+
+import traceback
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import HAS_BOTO3, camel_dict_to_snake_dict, boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def create_pipeline(client, name, role_arn, artifact_store, stages, version, module):
+    pipeline_dict = {'name': name, 'roleArn': role_arn, 'artifactStore': artifact_store, 'stages': stages}
+    if version:
+        pipeline_dict['version'] = version
+    try:
+        resp = client.create_pipeline(pipeline=pipeline_dict)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable create pipeline {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to create pipeline {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def update_pipeline(client, pipeline_dict, module):
+    try:
+        resp = client.update_pipeline(pipeline=pipeline_dict)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable update pipeline {0}: {1}".format(pipeline_dict['name'], to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to update pipeline {0}: {1}".format(pipeline_dict['name'], to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def delete_pipeline(client, name, module):
+    try:
+        resp = client.delete_pipeline(name=name)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable delete pipeline {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to delete pipeline {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def describe_pipeline(client, name, version, module):
+    pipeline = {}
+    try:
+        if isinstance(version, int) is int:
+            pipeline = client.get_pipeline(name=name, version=version)
+            return pipeline
+        else:
+            pipeline = client.get_pipeline(name=name)
+            return pipeline
+    except botocore.exceptions.ClientError as e:
+        return pipeline
```

Comment:
Instead of ignoring all ClientErrors here, check the error code:
```
except botocore.exceptions.ClientError as e:
    if e.response['Error']['Code'] == 'PipelineNotFoundException':
        return pipeline
    else:
        module.fail_json(....)
```

Reply:
Just adding a follow up comment - we've made a simpler way to do this with is_boto3_error_code() from ansible.module_utils.aws.core to catch specific exceptions:
```
try:
    ...
except is_boto3_error_code('PipelineNotFoundException'):
    return pipeline
except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:  # pylint: disable=duplicate-except
    module.fail_json_aws(e)
```